include/libdnf/module/module_item.hpp

```diff
@@ -0,0 +1,226 @@
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getArch()
+    std::string get_arch() const;
+
+    /// @return The "name:stream:version:context:arch" string.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getFullIdentifier()
+    std::string get_full_identifier() const;
+
+    /// @return The summary of this ModuleItem.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getSummary()
+    std::string get_summary() const;
+
+    /// @return The description of this ModuleItem.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getDescription()
+    std::string get_description() const;
+
+    // TODO(pkratoch): Translations for summary and description
+
+    /// @return The list of RPM NEVRAs in this module.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getArtifacts()
+    std::vector<std::string> get_artifacts() const;
+
+    /// @return Sorted list of RPM names that are demodularized.
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getDemodularizedRpms()
+    std::vector<std::string> get_demodularized_rpms() const;
+
+    /// @return The list of ModuleProfiles matched by name (possibly a globby pattern).
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getProfiles(const std::string &name)
+    std::vector<ModuleProfile> get_profiles(const std::string & name) const;
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getProfiles()
+    std::vector<ModuleProfile> get_profiles() const;
+
+    // TODO(pkratoch): Similar implementation as in ModuleItemContainer::getDefaultProfiles().
+    /// @return The default profiles.
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getDefaultProfile()
+    std::vector<ModuleProfile> get_default_profiles() const;
+
+    /// Get the `static_context` value (needed for proper behaviour of modular solver).
+    ///
+    /// @return The value of the `static_context` flag in modulemd. If `true`, the context string
+    ///         is static and represents a build and runtime configuration for this stream.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getStaticContext()
+    bool get_static_context() const;
+
+    /// @return The list of strings of modules requires (e.g. "nodejs", "nodejs:12", "nodejs:-11").
+    /// @param remove_platform When true, the method will not return requires with stream "platform"
+    //                         (default is false).
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getRequires(bool removePlatform=false)
+    std::vector<std::string> get_requires(bool remove_platform = false);
+
+    bool operator==(const ModuleItem & r) const;
+
+    ~ModuleItem();
```

Comment:
Please can you move constructors (not for this class but it is a general request), destructors and operator at the beginning of the class? See https://github.com/rpm-software-management/libdnf/blob/dnf-5-devel/include/libdnf/rpm/package_query.hpp